src/MChestOnMap.java

```diff
@@ -0,0 +1,194 @@
+            this.items.add(item, this.itemQuantity(item) + quantity);
+        } else {
+            // If item doesn't exist, add it to the chest
+            this.items.add(item, quantity);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Map.Pair<String, Integer> removeItem(String item, int quantity) {
+        assert item != null : "Violation of: item is not null";
+        assert quantity > 0 : "Violation of: quantity > 0";
+        assert this.containsItem(item) : "Violation of: item is in the chest";
+
+        int currentQuantity = this.itemQuantity(item);
+
+        if (currentQuantity <= quantity) {
+            Map.Pair<String, Integer> removedPair = this.items.remove(item);
+            return removedPair;
+        } else {
+            // Not gonna lie I didn't know how to bypass not making an instance
+            // of Map.Pair so that's why it's scuffed
+            this.items.remove(item);
+            this.items.add(item, currentQuantity - quantity);
+            Map.Pair<String, Integer> removedPair = this.items.remove(item);
+            return removedPair;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean containsItem(String item) {
+        assert item != null : "Violation of: item is not null";
+        return this.items.hasKey(item);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int itemQuantity(String item) {
+        assert item != null : "Violation of: item is not null";
+        if (this.items.hasKey(item)) {
+            return this.items.value(item);
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isFull() {
+        final int thousand = 1000;
+        return this.totalItems() >= thousand;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void clear() {
+        this.items = new Map2<>(); // Reset the chest to an empty state
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        String result = "MChest: ";
+        for (Map.Pair<String, Integer> pair : this.items) {
+            result += "[" + pair.key() + ", " + pair.value() + "] ";
+        }
+        return result.trim();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj instanceof MChest) {
+            return this.items.equals(((MChest) obj).getItems());
+        }
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        return this.items.hashCode();
+    }
+
+    // Implementing Standard<MChest> methods
+
+    // CHECKSTYLE: ALLOW THIS METHOD TO BE OVERRIDDEN
+    @Override
+    public MChest newInstance() {
+        return new MChestOnMap();
```

Comment:
This is probably fine, but take a peek at your previous projects to see how we implement this.